src/Umbraco.Infrastructure/Services/Implement/FileService.cs

```diff
@@ -342,27 +344,23 @@ public long GetScriptFileSize(string filepath)
                 template.Content = content;
             }
 
-
-
-
-            using (var scope = ScopeProvider.CreateScope())
+            using (IScope scope = ScopeProvider.CreateScope())
             {
-                var saveEventArgs = new SaveEventArgs<ITemplate>(template, true, evtMsgs, additionalData);
-                if (scope.Events.DispatchCancelable(SavingTemplate, this, saveEventArgs))
+                var savingEvent = new TemplateSavingNotification(template, eventMessages, additionalData);
+                if (scope.Notifications.PublishCancelable(savingEvent))
                 {
                     scope.Complete();
-                    return OperationResult.Attempt.Fail<OperationResultType, ITemplate>(OperationResultType.FailedCancelledByEvent, evtMsgs, template);
+                    return OperationResult.Attempt.Fail<OperationResultType, ITemplate>(OperationResultType.FailedCancelledByEvent, eventMessages, template);
                 }
 
                 _templateRepository.Save(template);
-                saveEventArgs.CanCancel = false;
-                scope.Events.Dispatch(SavedTemplate, this, saveEventArgs);
+                scope.Notifications.Publish(new TemplateSavedNotification(template, eventMessages, savingEvent.AdditionalData).WithStateFrom(savingEvent));
```

Comment:
I think we will need to do something similar to `WithStateFrom(savingEvent)` for the `AdditionalData`, otherwise that object is not shared between the `TemplateSavingNotification` and `TemplateSavedNotification` like it was before

Reply:
Nevermind, I see you are using that explict type when you create the `TemplateSavedNotification `